bgpd/bgp_routemap.c

```diff
@@ -548,6 +548,41 @@ struct route_map_rule_cmd route_match_ip_next_hop_prefix_list_cmd = {
 	route_match_ip_next_hop_prefix_list_compile,
 	route_match_ip_next_hop_prefix_list_free};
 
+/* `match ip next-hop type <blackhole>' */
+
+static route_map_result_t
+route_match_ip_next_hop_type(void *rule, const struct prefix *prefix,
+			     route_map_object_t type, void *object)
+{
+	struct bgp_info *bgp_info;
+
+	if (type == RMAP_BGP && prefix->family == AF_INET) {
+		bgp_info = (struct bgp_info *)object;
+		if (!bgp_info || !bgp_info->attr)
+			return RMAP_DENYMATCH;
+
+		if (bgp_info->attr->nexthop.s_addr == INADDR_ANY
```

Comment:
Adding a comment here would be helpful explaining the logic for this check. Without the knowledge of #1388, it may be little difficult for the user to follow.